csharp/test/Apache.Arrow.Tests/TypeTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Apache.Arrow.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Apache.Arrow.Tests
+{
+    public class FieldComparer
+    {
+        public static bool Equals(Field f1, Field f2)
+        {
+            if (f2 != null && f1 != null && f1.Name == f2.Name && f1.IsNullable == f2.IsNullable &&
```

Comment:
Question for the crowd: Are names case sensitive or insensitive in Apache Arrow?

Reply:
The cpp implementation under type.cc suggests that it is case sensitive.

Reply:
Having a comparer here seems odd if it is just case sensitive.

https://github.com/apache/arrow/blob/6fb4cd0e00fb17aaf80113280ee642494ff3296e/csharp/src/Apache.Arrow/Schema.cs#L58-L65

Reply:
In C++ they are case sensitive. This isn't prescriptive, though

Reply:
Shouldn't we default to InvariantCulture? Thoughts?

Reply:
Honestly, I would use `StringComparer.Ordinal` by default. That's what is used when you do `string1 == string2` in C#.